src/main/java/org/testng/internal/ParametersTest.java

```diff
@@ -0,0 +1,31 @@
+package org.testng.internal;
+
+import java.lang.reflect.Method;
+
+import org.testng.Assert;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+import org.testng.internal.Parameters.FilterOutInJectedTypesResult;
+import org.testng.internal.annotations.JDK15AnnotationFinder;
+import org.testng.xml.XmlTest;
+
+/**
+ * Provide test for package visible methods in org.testng.internal.Parameters 
+ *
+ */
+public class ParametersTest {
+    
+    @Test
+    @Parameters({ "testdata" })
+    public void filterOutInJectedTypesFromOptionalValuesTest(XmlTest xmlTest, @Optional("optionaltestdata") String testdata) {
+      JDK15AnnotationFinder finder = new JDK15AnnotationFinder(null);
+      Method curMethod = new Object() {}.getClass().getEnclosingMethod();
```

Comment:
`Object[].class.getEnclosingMethod()`

Reply:
No, it doesn't work like this.  NPE would be thrown.

Reply:
True, I read it too quickly.